D/D.sublime-syntax

```diff
@@ -8,339 +8,2032 @@ file_extensions:
+    - include: floating-point-opt
+    - include: not-whitespace-illegal
+  floating-point-opt:
+    - match: '\b(0[xX])({{hex_or_under}}*\.?{{hex_or_under}}+)([pP][+\-]?[0-9]+)({{float_suffix}})?\b'
+      captures:
+        1: punctuation.definition.numeric.hex.d
+        2: constant.numeric.float.d
+        3: constant.numeric.float.d
+        4: storage.type.float.d
+      pop: true
+    - match: '\b(0|[1-9][0-9_]*)({{integer_float_suffix}})\b'
+      captures:
+        1: constant.numeric.float.d
+        2: storage.type.float.d
+      pop: true
+    - match: '\b(0[bB])([01][01_]*)({{integer_float_suffix}})\b'
+      captures:
+        1: punctuation.definition.numeric.binary.d
+        2: constant.numeric.float.d
+        3: storage.type.float.d
+      pop: true
+    - match: '\b(0[xX])({{hex_char}}{{hex_or_under}}*)({{integer_float_suffix}})\b'
+      captures:
+        1: punctuation.definition.numeric.hex.d
+        2: constant.numeric.float.d
+        3: storage.type.float.d
+      pop: true
+    - match: '\b([0-9][0-9_]*\.[0-9][0-9_]*)({{float_exponent}})?({{float_suffix}})?\b'
+      captures:
+        1: constant.numeric.float.d
+        2: constant.numeric.float.d
+        3: storage.type.float.d
+      pop: true
+    - match: '\b([0-9][0-9_]*\.(?!\.|\s*[a-zA-Z]))({{float_suffix}})?'
+      captures:
+        1: constant.numeric.float.d
+        2: storage.type.float.d
+      pop: true
+    - match: '(\.[0-9][0-9_]*)({{float_exponent}})?({{float_suffix}})?\b'
+      captures:
+        1: constant.numeric.float.d
+        2: constant.numeric.float.d
+        3: storage.type.float.d
+      pop: true
+    - match: '\b([0-9][0-9_]*)({{float_exponent}})({{float_suffix}})?\b'
+      captures:
+        1: constant.numeric.float.d
+        2: constant.numeric.float.d
+        3: storage.type.float.d
+      pop: true
+
+  character-in:
+    - match: '{{character_lookahead}}'
+      push: character
+  character:
+    - match: \'(?:({{escape_sequence}})|[^\\'])\'
+      scope: string.quoted.single.d
+      captures:
+        1: constant.character.escape.d
+      pop: true
+    - match: \'[^']*\'
+      scope: invalid.illegal.d
+      pop: true
+    - include: not-whitespace-illegal-pop
+
+  string-in:
+    - match: '{{string_lookahead}}'
+      push: string
+  string:
+    - include: string-opt
+    - include: not-whitespace-illegal
+  string-opt:
+    # Regular string
+    - match: '"'
+      scope: punctuation.definition.string.begin.d
+      set:
+        - meta_include_prototype: false
+        - meta_scope: string.quoted.double.d
+        - match: '(")({{string_postfix}})'
+          captures:
+            1: punctuation.definition.string.end.d
+            2: storage.type.string.d
           pop: true
-        - match: \b(_|:)\b
+        - match: '{{escape_sequence}}'
+          scope: constant.character.escape.d
+        - match: \\.
+          scope: invalid.illegal.unknown-escape.d
+    # Wysiwyg string
+    - match: r"
```

Comment:
Does this still make sense to use `storage.type` considering a prefix never influences the type of the string and is only lexical, whereas the postfixes do actually change the type of the string literal?

Reply:
Iirc this question was raised for Python recently since `u` and `r` don't affect the string type, but it wasn't decided on what it should become. Maybe `storage.modifier`?

Reply:
At least in python there's also the `b` prefix which does affect the type, so I can see an argument for highlighting all prefixes the same. `rb''` would look a bit odd if `r` and `b` are highlighted differently.